opentasks-provider/src/main/java/org/dmfs/provider/tasks/processors/tasks/Instantiating.java

```diff
@@ -45,10 +47,14 @@
     /**
      * This is a field adapter for a pseudo column to indicate that the instances may need an update, even if no relevant value has changed. This is useful to
      * force an update of the sorting values when the local timezone has been changed.
+     * <p>
+     * TODO: get rid of it
      */
     private final static BooleanFieldAdapter<TaskAdapter> UPDATE_REQUESTED = new BooleanFieldAdapter<TaskAdapter>(
             "org.dmfs.tasks.TaskInstanceProcessor.UPDATE_REQUESTED");
 
+    private final static int INSTANCE_COUNT_LIMIT = 1000;
```

Comment:
Is this number about how many recurring instance will be created? If yes, isn't this way too much? I mean that I, as a user, would be fine with having let's say the next 30 occurrence available for me in the app.

Reply:
That depends, if you have a daily recurring task 30 might not be sufficient. AFAIK, Android's calendar provider has a limit of 2000. Let's check if that becomes a bottleneck. The iterator is quite fast and can do that in just a few ms. Instance creation might take a little longer though. I also was planning to add a date limit (i.e. don't expand further than 10 years from now). But that's merely an optimization. The good thing about how this is done is, we can change these limits later on without breaking anything.